extensions/lifecycle/lifecycleConsumer/LifecycleConsumer.js

```diff
@@ -0,0 +1,101 @@
+     * @param {String} lcConfig.objectTasksTopic - lifecycle object topic name
+     * @param {Object} lcConfig.consumer - kafka consumer object
+     * @param {String} lcConfig.consumer.groupId - kafka consumer group id
+     * @param {Integer} lcConfig.consumer.retryTimeoutS - number of seconds
+     *  before giving up retries of an entry lifecycle action
+     * @param {Integer} lcConfig.consumer.concurrency - number of max allowed
+     *  concurrent operations
+     * @param {Object} s3Config - S3 configuration
+     * @param {Object} s3Config.s3 - s3 endpoint configuration object
+     * @param {Object} authConfig - authentication info on source
+     */
+    constructor(zkConfig, lcConfig, s3Config, authConfig) {
+        super();
+        this.zkConfig = zkConfig;
+        this.lcConfig = lcConfig;
+        this.s3Config = s3Config;
+        this.authConfig = authConfig;
+        this._consumer = null;
+
+        this.logger = new Logger('Backbeat:Lifecycle:ObjectConsumer');
+
+        // global variables
+        // TODO: for SSL support, create HTTPS agents instead
+        this.httpAgent = new http.Agent({ keepAlive: true });
+
+        // this.taskScheduler = new ReplicationTaskScheduler(
+        //     (ctx, done) => ctx.task.processQueueEntry(ctx.entry, done));
+    }
+
+
+    /**
+     * Start kafka consumer. Emits a 'ready' event when
+     * consumer is ready.
+     *
+     * @return {undefined}
+     */
+    start() {
+        this._consumer = new BackbeatConsumer({
+            zookeeper: {
+                connectionString: this.zkConfig.connectionString,
+            },
+            topic: this.lcConfig.objectTasksTopic,
+            groupId: this.lcConfig.consumer.groupId,
+            concurrency: this.lcConfig.consumer.concurrency,
+            queueProcessor: this.processKafkaEntry.bind(this),
+            // fetchMaxBytes: CONSUMER_FETCH_MAX_BYTES,
+        });
+        this._consumer.on('error', () => {});
+        this._consumer.subscribe();
+        this.logger.info('queue processor is ready to consume ' +
+                            'replication entries');
+        return this.emit('ready');
+    }
+
+
+    /**
+     * Proceed to the lifecycle action of an object given a kafka
+     * object lifecycle queue entry
+     *
+     * @param {object} kafkaEntry - entry generated by the queue populator
+     * @param {function} done - callback function
+     * @return {undefined}
+     */
+    processKafkaEntry(kafkaEntry, done) {
+        let entryData;
+        try {
+            entryData = JSON.parse(kafkaEntry.value);
+        } catch (err) {
+            this.logger.error('error processing lifecycle object entry',
+                { error: err });
+            return process.nextTick(() => done(errors.InternalError));
+        }
+        const task = new LifecycleObjectTask();
+        const q = async.queue(task.processQueueEntry);
+        return q.push(entryData, done);
```

Comment:
Wondering if it worth including calling the `drain` method on the queue to log any errors from `LifecycleObjectTask.processQueueEntry`.